### DISCUSSION TITLE
Bug report

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#242] [Modeling] [answered]
# class Delay in discrete.py

When there is multiple delay time, the class Delay cannot work well.

I added a variable ``self.fd = Delay(...)`` in a model, but there is error.

In the discrete.py, line 1261 - 1266
```python
                if dae_t - self.t[0] > self.delay:
                    t_interp = dae_t - self.delay
                    idx = np.argmax(self.t >= t_interp) - 1
                    v_interp = interp_n2(t_interp,
                                         self.t[idx:idx+2],
                                         self._v_mem[:, idx:idx + 2])
```
The ``self.delay`` is an array, which means the comparison cannot be operated. Also, the memory variable ``self._v_mem`` should be extended to multiple dimensions. I'm not sure how to fix this.

Regards,
Jinning

**Comment by cuihantao** [comment, marked as answer]
This seems to be a known issue. See the doc string in Delay: self.delay needs to be a scalar. 

I think you can implement a DelayVec for the desired function. You can use for loops in check_var or check_eq and design suitable data structure. To store the values within the delay window, using a dictionary with the keys being the idxes and the values being a list of values can be a good choice.

**Reply by jinningwang** [reply]
Thanks for your reply, I'll take a try.
